azure-cs-botservice/Program.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright 2016-2019, Pulumi Corporation.  All rights reserved.
+
+using Pulumi;
+using Pulumi.Azure.AppService;
+using Pulumi.Azure.AppService.Inputs;
+using Pulumi.Azure.Core;
+using Pulumi.Azure.Storage;
+using Pulumi.AzureAD;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+class Program
+{
+    static Task<int> Main(string[] args)
+    {
+        return Deployment.RunAsync(() =>
+        {
+            var config = new Config();
+            var botName = config.Require("botName");
+
+            var resourceGroup = new ResourceGroup("botservice-rg");
+
+            var storageAccount = new Pulumi.Azure.Storage.Account("sa", new Pulumi.Azure.Storage.AccountArgs
+            {
+                ResourceGroupName = resourceGroup.Name,
+                AccountReplicationType = "LRS",
+                AccountTier = "Standard"
+            });
+
+            var appServicePlan = new Plan("asp", new PlanArgs
+            {
+                ResourceGroupName = resourceGroup.Name,
+                Kind = "App",
+                Sku = new PlanSkuArgs
+                {
+                    Tier = "Basic",
+                    Size = "B1"
+                },
+            });
+
+            var container = new Container("zips", new ContainerArgs
+            {
+                StorageAccountName = storageAccount.Name,
+                ContainerAccessType = "private",
+            });
+
+            var blob = new ZipBlob("zip", new ZipBlobArgs
+            {
+                StorageAccountName = storageAccount.Name,
+                StorageContainerName = container.Name,
+                Type = "block",
+                Content = new FileArchive("bot/publish")
+            });
+
+            var codeBlobUrl = SharedAccessSignature.SignedBlobReadUrl(blob, storageAccount);
+
+            var appInsights = new Pulumi.Azure.AppInsights.Insights("ai", new Pulumi.Azure.AppInsights.InsightsArgs
+            {
+                ApplicationType = "web",
+                ResourceGroupName = resourceGroup.Name
+            });
+
+            var appInsightApiKey = new Pulumi.Azure.AppInsights.ApiKey("ai", new Pulumi.Azure.AppInsights.ApiKeyArgs
+            {
+                ApplicationInsightsId = appInsights.Id,
+                ReadPermissions = "api",
+            });
+
+            var luis = new Pulumi.Azure.Cognitive.Account("cs", new Pulumi.Azure.Cognitive.AccountArgs
+            {
+                Kind = "CognitiveServices", // includes LUIS
+                ResourceGroupName = resourceGroup.Name,
+                Sku = new Pulumi.Azure.Cognitive.Inputs.AccountSkuArgs() { Name = "S0", Tier = "Standard" }
```

Comment:
Nit: you don't have to add `() after the constructor name. Saves 2 characters :)

Reply:
good